R/Private.R

```diff
@@ -115,3 +115,157 @@ nullToEmpty <- function(x) {
   x[is.null(x)] <- ""
   return(x)
 }
+
+
+
+makeDataExportable <- function(x,
+                               tableName,
+                               minCellCount = 5,
+                               databaseId = NULL) {
+  ParallelLogger::logTrace(paste0(" - Ensuring data is exportable: ", tableName))
+  if (is.null(x) || nrow(x) == 0) {
+    ParallelLogger::logTrace("  - Object has no data.")
+  }
+  
+  resultsDataModel <- getResultsDataModelSpecifications()
+  
+  if (!is.null(databaseId)) {
+    x <- x %>%
+      dplyr::mutate(databaseId = !!databaseId)
+  }
+  
+  fieldsInDataModel <- resultsDataModel %>%
+    dplyr::filter(.data$tableName == !!tableName) %>%
```

Comment:
what is the `!!tableName` for. The double negation implies that it is the tableName so `dplyr::filter(.data$tableName == !!tableName)` is functionally identical.

Tbh I'm not sure why !!tableName doesn't just bomb (!<string>) is clearly undefined; so R's operator precedence is just weird here.